src/fondant/component_spec.py

```diff
@@ -230,85 +229,159 @@ class KubeflowComponentSpec:
+        )
+        output_definitions = {
+            "artifacts": {
+                "output_manifest_path": {
+                    "artifactType": {
+                        "schemaTitle": "system.Artifact",
+                        "schemaVersion": "0.0.1",
+                    },
                     "description": "Path to the output manifest",
-                    "type": "String",
                 },
-            ],
-            "implementation": {
-                "container": {
-                    "image": fondant_component.image,
-                    "command": [
-                        "fondant",
-                        "execute",
-                        "main",
-                        "--input_manifest_path",
-                        {"inputPath": "input_manifest_path"},
-                        "--metadata",
-                        {"inputValue": "metadata"},
-                        "--component_spec",
-                        {"inputValue": "component_spec"},
-                        "--input_partition_rows",
-                        {"inputValue": "input_partition_rows"},
-                        "--cache",
-                        {"inputValue": "cache"},
-                        *cls._dump_args(fondant_component.args.values()),
-                        "--output_manifest_path",
-                        {"outputPath": "output_manifest_path"},
-                    ],
+            },
+        }
+
+        specification = {
+            "components": {
+                "comp-"
+                + cleaned_component_name: {
+                    "executorLabel": "exec-" + cleaned_component_name,
+                    "inputDefinitions": input_definitions,
+                    "outputDefinitions": output_definitions,
+                },
+            },
+            "deploymentSpec": {
+                "executors": {
+                    "exec-"
+                    + cleaned_component_name: {
+                        "container": {
+                            "args": [
+                                "--input_manifest_path",
+                                "{{$.inputs.artifacts['input_manifest_path'].uri}}",
+                                "--metadata",
+                                "{{$.inputs.parameters['metadata']}}",
+                                "--component_spec",
+                                "{{$.inputs.parameters['component_spec']}}",
+                                "--input_partition_rows",
+                                "{{$.inputs.parameters['input_partition_rows']}}",
+                                "--cache",
+                                "{{$.inputs.parameters['cache']}}",
+                                *cls._dump_args(fondant_component.args.values()),
+                                "--output_manifest_path",
+                                "{{$.outputs.artifacts['output_manifest_path'].uri}}",
+                            ],
+                            "command": ["python3", "main.py"],
```

Comment:
Shouldn't we use here `fondant execute main` instead?

Reply:
ah really ? Where do we do that ? I just took over the way we did it for kfpv1.

Reply:
We changed while resolving this: #362

Reply:
See comment above